go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -875,76 +845,82 @@ func (sess *Session) setForeignKeyChecksSessionVar(ctx *sql.Context, key string,
 	return sess.Session.SetSessionVariable(ctx, key, value)
 }
 
-func (sess *Session) setWorkingSessionVar(ctx *sql.Context, value interface{}, dbName string) error {
-	valStr, isStr := value.(string) // valStr represents a root val hash
-	if !isStr || !hash.IsValid(valStr) {
-		return doltdb.ErrInvalidHash
-	}
+func (sess *Session) resolveVersionKey(ctx *sql.Context, key, value string) (ws ref.WorkingSetRef, db string, err error) {
```

Comment:
I don't love the behavior here, where if there are two branches with the same HEAD commit (common), then setting db_head dies.

Needs to be fixed soon, preferably by making db_head read-only. Add some TODOs here